wsdl-core/src/main/java/io/ballerina/wsdl/core/OperationProcessor.java

```diff
@@ -0,0 +1,263 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.wsdl.core;
+
+import io.ballerina.wsdl.core.recordgenerator.ballerinair.BasicField;
+import io.ballerina.wsdl.core.recordgenerator.ballerinair.ComplexField;
+import io.ballerina.wsdl.core.recordgenerator.ballerinair.Field;
+import io.ballerina.wsdl.core.recordgenerator.ballerinair.XMLNSAttribute;
+import io.ballerina.wsdl.core.wsdlmodel.SOAPVersion;
+import io.ballerina.wsdl.core.wsdlmodel.WSDLHeader;
+import io.ballerina.wsdl.core.wsdlmodel.WSDLOperation;
+import io.ballerina.wsdl.core.wsdlmodel.WSDLPart;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class OperationProcessor {
+    private final WSDLOperation wsdlOperation;
+    private final List<WSDLOperation> processedOperations;
+    private final SOAPVersion soapVersion;
+    private Map<String, String> uriToPrefixMap = new HashMap<>();
+    private Map<String, String> prefixToUriMap = new HashMap<>();
+
+    OperationProcessor(WSDLOperation wsdlOperation, List<WSDLOperation> processedOperations, SOAPVersion soapVersion) {
```

Comment:
Any reason to use the default access modifier here ? If we don't have any specific reason we can use `protected` instead.